lib/model/basket_item.rb

```diff
@@ -0,0 +1,34 @@
+module Shop
+  class BasketItem
+    attr_reader :product_id, :quantity
+
+    ProductNotFound = Class.new(StandardError)
+    WrongQuantityRequested = Class.new(StandardError)
+    NoSuchItem = Class.new(StandardError)
+
+    def initialize(product_id:, quantity: 1)
+      @product_id = product_id
+      @quantity = quantity
+
+      raise ArgumentError unless quantity.is_a?(Integer)
+      raise ArgumentError unless quantity > 0
+    end
+
+    def fetch_product
+      FetchProduct.new.call(product_id)
+    end
+
+    def unit_price
+      product = fetch_product
```

Comment:
I agreed with you initially, but come to think of it, this method is also used in fetch_basket_items.rb (in lib/services/), so I think it's better to leave it as it is. What do you think?